src/gr11review/part2/Utility.java

```diff
@@ -1,9 +1,269 @@
+
+  // declare variables
+  int numFound = 0;
+  int k = 0;
+
+  // if the length of the inner array is 0, it will always return true
+  if(inner.length == 0) {
+     return true;
+  }
+  
+  // determine if the integers in the outer array are present in the inner array. 
+  // if they are the present, return true. otherwise return false. 
+  for(int i = 0; i < outer.length; i++) {
+     if(outer[i] == inner[k]) {
+        numFound++;
+        k++;
+     } else if(outer[i] > inner[k]) {
+        return false;
+     }
+     
+     if(numFound == inner.length)
+        return true;
+  }
+
+  // if nothing is found, return false
+  return false;
+}
+  
+  /**
+  * This is a program to compare the value of left and side, if they are equal or not
+  * @author: Zyaire, Qu
+  */
+  public static boolean canBalance(int[] nums){//array 4
+    int ls = 0;//initialize the values
+    int rs = 0;
+    int count;
+    int countTwo;
+    for(count = 0; count < nums.length; count++){//here is the loop
+      ls += nums[count];//adding the left side
+      rs = 0;//reset because it will be so big
+      for(countTwo = nums.length-1; countTwo > count; countTwo--){//adding the right side
+        rs += nums[countTwo]; 
+      }
+      if(ls == rs){//check if correct
+        return true;
+      }
+    }
+    return false;
+  }
+  
+/**
+* This program adds values top and left in a 2D array to get the value of the next spot on a table
+* @author: Douglas, Lau
+*/
+public static void pascalTri(int i, int j) throws IOException {// Array 5 - Two Dimensional
+
+  // declare variables
+  int[][] num;
+
+  // determine the numbers in the Pascal triangle
+  num = new int[i][j];
+  for (int countH = 0; countH < i; countH++) {
+    for (int countV = 0; countV < j; countV++) {
+      if (countV == 0 || countH == 0) {
+        num[countH][countV] = 1;
+      }
+       else {
+        num[countH][countV] = num[countH][countV - 1] + num[countH - 1][countV];
+      }
+    }
+  }
+
+  // display the numbers in pascalOut.textFile
+   BufferedWriter myWriter = null;
+   myWriter = new BufferedWriter(new FileWriter("pascalOut.txt"));
+   for (int countH = 0; countH < i; countH++) {
+     for (int countV = 0; countV <= j; countV++) {
+       if (countV == j) {
+         myWriter.write("\n");
+       }
+       else {
+         myWriter.write(num[countH][countV] + "");
+         if (countV < j - 1) {
+           myWriter.write(",");
+         }
+       }
+     }
+   }
+   myWriter.close();
+  }
+  
+  /**
+  * This is a program that prints a two dimentional array's diagonal
+  * @author: Zyaire, Qu
+  */
+  public static void diagonal(int n) throws IOException{//array6
+    int count;
+    int countC;
+    //PrintWriter output = new PrintWriter(new FileWriter("src/gr11review/part2/diagonalOut.txt", true)); 
+    PrintWriter output = new PrintWriter(new FileWriter("diagonalOut.txt", true)); 
```

Comment:
the parameter for `new FileWriter("diagonalOut.txt", true)` should not be set to `true`.  this causes and append to existing file contents.  should be `false` or not given a value at all